File: verify.py
from playwright.sync_api import sync_playwright
import time

def take_screenshot():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.set_viewport_size({"width": 800, "height": 800}) # Square ratio close to image

        # Go to local server
        page.goto('http://localhost:8000')

        # Wait for WebGL to render
        time.sleep(3)

        # Take screenshot
        page.screenshot(path='/home/jules/verification/grid.png')
        print("Screenshot saved to /home/jules/verification/grid.png")

        browser.close()

if __name__ == "__main__":
    import os
    os.makedirs('/home/jules/verification', exist_ok=True)
    take_screenshot()

File: verify.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  // Navigate to the local app
  await page.goto('http://localhost:8000');

  // Wait for initial render
  await page.waitForTimeout(3000);
  console.log('Initial render complete');

  // Click Electron
  await page.click('#btn-electron');
  await page.waitForTimeout(3000);
  console.log('Electron rendered');

  // Click Proton
  await page.click('#btn-proton');
  await page.waitForTimeout(3000);
  console.log('Proton rendered');

  // Click Photon
  await page.click('#btn-photon');
  await page.waitForTimeout(3000);
  console.log('Photon rendered');

  // Check for any console errors
  page.on('console', msg => {
      if (msg.type() === 'error') {
          console.error(`Page Error: ${msg.text()}`);
      }
  });

  await browser.close();
})();
